lib/associations/belongs-to-many.js

```diff
@@ -298,6 +298,46 @@ var BelongsToMany = function(source, target, options) {
 
 util.inherits(BelongsToMany, Association);
 
+BelongsToMany.prototype.get = function(instances, options) {
+  var throughModel = this.throughModel;
+
+  var throughOptions = _.assign({
+    where: _.set({}, this.foreignKey, {$in: _.pluck(instances, this.source.primaryKeyAttribute)}),
+    logging: options.logging
+  });
+
+  var association = this;
+
+  return throughModel.findAll(throughOptions).then(function (throughs) {
+    var otherKey = association.otherKey
+      , targetIds = _.chain(throughs).pluck(otherKey).uniq().value();
+
+    var targetOptions = _.chain({
+      where: _.set({}, association.target.primaryKeyAttribute, {$in: targetIds})
+    })
+        .assign(association.target.$optClone(options))
+        .omit(['include'])
+        .value();
+
+    return Utils.Promise.join(throughs, _.isEmpty(targetIds) ? null : association.target.findAll(targetOptions));
```

Comment:
What you said about only having a single `right_id` per `left_id` is true in the case of a `hasMany`, but will mean that the source can only be associated with a single target in a `belongsToMany`, which defeats the purpose.
BTW, the `UNION ALL` trick will work, but is essentially the same as running a query per source, which is very inefficient (but can solve the case with the `limit: ...`).

Would love a second opinion since I'm not completely certain about what I'm saying...